packages/e2e-tests/specs/editor/various/links.test.js

```diff
@@ -81,6 +81,30 @@ describe( 'Links', () => {
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 	} );
 
+	it( 'will not pre-populate URL input if selected text is not a link', async () => {
+		// Create a block with some text
+		await clickBlockAppender();
+		await page.keyboard.type( 'This: is not a link' );
+
+		// Select some text
+		await pressKeyWithModifier( 'shiftAlt', 'ArrowLeft' );
+
+		// Click on the Link button
+		await page.click( 'button[aria-label="Link"]' );
+
+		// Wait for the URL field to auto-focus
+		await waitForAutoFocus();
+
+		const urlInputValue = await page.evaluate(
+			() =>
+				document.querySelector(
+					'.components-popover__content .block-editor-link-control .block-editor-url-input__input'
+				).value
+		);
```

Comment:
I'm not keen on relying on developer focused APIs for these kind of lookups. The entire CSS cascade you've specified could change at any level and the test would break. If possible, I think we can make it more resilient to change but focusing on selecting this via user facing APIs such as aria roles or text content.

For example you could explore using the `aria-label="URL"` to select the input. This much less likely to change.

Reply:
Great point. I'll update this.